Assets/Player/PsychosisBar.cs

```diff
@@ -0,0 +1,39 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PsychosisBar : MonoBehaviour
+{
+    public Slider slider;
+    private float _timer;
+    public float currentValue;
+    public float coolDownTime;
+    // Start is called before the first frame update
+    void Start()
+    {
+        slider.value = currentValue;
+    }
+    // Update is called once per frame
+    void Update()
+    {
+        _timer += Time.deltaTime;
+        //if the time is bigger than cooldown, it decreases the bar value and restarts the cooldown
+        if (_timer >= coolDownTime)
```

Comment:
Make the cooldown go down progressively instead of with steps
Think of it like `currentValue -= Time.deltaTime * speed`